src/components/views/elements/Reply.js

```diff
@@ -0,0 +1,214 @@
+            events: [],
+
+            // The latest loaded event which has not yet been shown
+            loadedEv: null,
+            // Whether the component is still loading more events
+            loading: true,
+
+            // Whether as error was encountered fetching a replied to event.
+            err: false,
+        };
+
+        this.onQuoteClick = this.onQuoteClick.bind(this);
+
+        this.unmounted = false;
+    }
+
+    componentWillMount() {
+        this.room = this.getRoom(this.props.parentEv.getRoomId());
+        this.initialize();
+    }
+
+    componentWillUnmount() {
+        this.unmounted = true;
+    }
+
+    async initialize() {
+        const {parentEv} = this.props;
+        const inReplyTo = Reply.getInReplyTo(parentEv);
+
+        const ev = await this.getEvent(this.room, inReplyTo['event_id']);
+        if (this.unmounted) return;
+
+        if (ev) {
+            this.setState({
+                events: [ev],
+            }, this.loadNextEvent);
+        } else {
+            this.setState({err: true});
+        }
+    }
+
+    async loadNextEvent() {
+        const ev = this.state.events[0];
+        const inReplyTo = Reply.getInReplyTo(ev);
+
+        if (!inReplyTo) {
+            if (this.unmounted) return;
+            this.setState({
+                loading: false,
+            }, this.props.onWidgetLoad);
+            return;
+        }
+
+        const loadedEv = await this.getEvent(this.room, inReplyTo['event_id']);
+        if (this.unmounted) return;
+
+        if (loadedEv) {
+            this.setState({loadedEv}, this.props.onWidgetLoad);
+        } else {
+            this.setState({err: true}, this.props.onWidgetLoad);
+        }
+    }
+
+    getRoom(id) {
+        const cli = MatrixClientPeg.get();
+        if (id[0] === '!') return cli.getRoom(id);
+
+        return cli.getRooms().find((r) => {
+            return r.getAliases().includes(id);
+        });
+    }
+
+    async getEvent(room, eventId) {
+        const event = room.findEventById(eventId);
+        if (event) return event;
+
+        try {
+            await MatrixClientPeg.get().getEventTimeline(room.getUnfilteredTimelineSet(), eventId);
+        } catch (e) {
+            return;
+        }
+        return room.findEventById(eventId);
+    }
+
+    onQuoteClick() {
+        const events = [this.state.loadedEv].concat(this.state.events);
+
+        this.setState({
+            loadedEv: null,
+            events,
+        }, this.loadNextEvent);
+
+        dis.dispatch({action: 'focus_composer'});
+    }
+
+    static getInReplyTo(ev) {
+        if (ev.isRedacted()) return;
+
+        const mRelatesTo = ev.getWireContent()['m.relates_to'];
+        if (mRelatesTo && mRelatesTo['m.in_reply_to']) {
+            const mInReplyTo = mRelatesTo['m.in_reply_to'];
+            if (mInReplyTo['event_id']) return mInReplyTo;
+        }
+    }
+
+    static getRelationship(ev) {
```

Comment:
I'd rename this to something specific to relates_to. Relationship isn't really a term defined at the protocol level.